pkg/api/v1beta3/conversion.go

```diff
@@ -20,11 +20,2638 @@ import (
 	"fmt"
 
 	newer "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/resource"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/conversion"
 )
 
 func init() {
+	err := newer.Scheme.AddConversionFuncs(
```

Comment:
Yes - those are autogenerated functions.
I agree that eventually this should be done how you described it (or similarly). But I don't think it's not a big problem now, because I'm focused now on v1beta3 where we don't have manually written conversion functions. But I'm going to work more on it next week.

I've added AddGeneratedConversionFunctions() method to scheme and I'm using it - I hope that would be enough for you for now.
BTW - I would like it to be merged soon (I will be working on it more right after this PR) to unblock other experiments that Filip is also doing in the performance area.